plugin.go

```diff
@@ -1,16 +1,75 @@
 package seabird
 
 import (
-	"github.com/belak/go-plugin"
+	"fmt"
+
+	"github.com/gobwas/glob"
+	"github.com/unknwon/com" //nolint:misspell
 )
 
-var plugins = plugin.NewRegistry()
+type PluginFactory func(b *Bot) error
+
+var plugins = make(map[string]PluginFactory)
 
 // RegisterPlugin registers a PluginFactory for a given name. It will
 // panic if multiple plugins are registered with the same name.
-func RegisterPlugin(name string, factory interface{}) {
-	err := plugins.Register(name, factory)
-	if err != nil {
-		panic(err.Error())
+func RegisterPlugin(name string, factory PluginFactory) {
+	if _, ok := plugins[name]; ok {
+		panic(fmt.Sprintf("Plugin %q registered multiple times", name))
+	}
+
+	plugins[name] = factory
+}
+
+func matchingPlugins(rawWhitelist, rawBlacklist []string) ([]string, error) {
```

Comment:
Why do we want the blacklist here if it's not used? Also, is it clear in the documentation that you can use glob matching for plugins?

Reply:
The blacklist is here because it was copied wholesale from go-plugin. I can remove the blacklist since we don't use it.

Reply:
Also, yes, I'll add some basic docs for that somewhere